packages/core/src/browser/tree/tree.ts

```diff
@@ -29,6 +29,10 @@ export interface Tree extends Disposable {
      * Setting a root node refreshes the tree.
      */
     root: TreeNode | undefined;
+    /**
+     * list of filtered by search nodes.
+     */
+    filteredNodes?: ReadonlyArray<Readonly<TreeNode>>;
```

Comment:
Could you elaborate on why its needed here?

Reply:
Removed that since changes in importing fixed the circular dependency :)